python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -829,8 +790,12 @@ def _process_auto_materialize_tick_generator(
                                 error=None,
                                 timestamp=evaluation_time.timestamp(),
                                 end_timestamp=None,
-                            ),
+                            )._replace(
+                                # make sure to override the evaluation id to stay on the previous value
+                                auto_materialize_evaluation_id=latest_tick.automation_condition_evaluation_id
+                            )
                         )
+                    # otherwise, tick completed normally, no need to do anything
                 else:
                     # (The evaluation IDs not matching indicates that the tick failed or crashed before
                     # the cursor could be written, so no runs have been launched and it's safe to
```

Comment:
this comment isn't quite accurate anymore right? The "The evaluation IDs not matching" part

Reply:
it's mostly accurate, just needs to be moved up to the place where we're doing the evaluation id comparison -- updated!